### PR TITLE
Fix cargo shim argument placement

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- ensure the publish-check cargo wrapper injects `--all-features` before any `--` separator so Cargo always receives the flag
- add unit coverage for the shim to cover separators, duplicates, and common global option scenarios

## Testing
- make test

------
https://chatgpt.com/codex/tasks/task_e_6909a3ec8d70832299d6a4dc1ca22261

## Summary by Sourcery

Enforce correct placement of the --all-features flag in the publish-check cargo shim and validate behavior with unit tests.

Enhancements:
- Inject --all-features before any -- separator or append it when no separator is present
- Preserve existing --all-features flags and skip injection for non-target commands

Tests:
- Add unit tests covering flag insertion before and after separators, duplicate handling, non-target commands, and global option scenarios

## Summary by CodeRabbit

* **Chores**
  * Added a CLI wrapper that ensures targeted cargo commands are always run with --all-features.

* **Tests**
  * Added unit tests for the argument rewriting logic and integration tests exercising the CLI behavior with a fake cargo executable.

### CODE REVIEW

File: tests/unit/test_cargo_shim.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    result = shim.rewrite_args(["test", "--", "--test-threads", "1"])
+    assert result == ["test", "--all-features", "--", "--test-threads", "1"]
+
+
+def test_appends_flag_when_no_separator() -> None:
+    shim = load_cargo_shim()
+    result = shim.rewrite_args(["check"])
+    assert result == ["check", "--all-features"]
+
+
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/unit/test_cargo_shim.py]: Has this now been resolved?

**Comment by leynos** [comment]
@coderabbitai have the following now been resolved?

- Consider adding an integration test that runs the shim via subprocess to validate real CLI behavior beyond the rewrite_args function.
- Add tests for other Cargo commands (e.g. 'bench' or 'clippy') that also support --all-features to ensure the shim injects flags consistently.
